src/ast/converter.js

```diff
@@ -1199,25 +1199,15 @@ export default class Converter {
     return AConst(StringValue(value));
   }
 
-  GetDate = (date, options, isDateTime) => {
-    date = date || new Date().toISOString();
-
-    if (!isDateTime) {
-      // the `date` value comes in as the string "2017-11-12 23:59:59". We want it to be interpreted as UTC for the
-      // purposes of the SQL query generation. So we convert the local timestamp to a UTC one. We don't care if it's
-      // in a different timezone, we just need to make sure the date component of the timestamp is identical to the
-      // value stored in the date field. We are effectively disregarding the time component of the timestamp.
-      return moment(date.replace(' ', 'T') + 'Z').utc();
-    }
-
+  GetDate = (dateString, options) => {
     const timeZone = (options && options.timeZone) || moment.tz.guess();
 
-    return moment.tz(date, timeZone);
+    return moment.tz(dateString ?? new Date().toISOString(), timeZone);
   }
 
-  ConvertDateValue = (date) => {
+  ConvertDateValue = (expression, date) => {
     if (date) {
-      return date.clone().toISOString();
+      return expression.column.isDateTime ? date.toISOString() : date.format('YYYY-MM-DD');
```

Comment:
For timestamps/datetime, format as ISO 8601, for `date`'s, we return the `YYYY-MM-DD` local date. This allows the definition of "today" (and all date math/conversion) for dates to depend on the timezone passed into this library.